packages/-ember-data/tests/integration/identifiers/configuration-test.ts

```diff
@@ -328,7 +361,10 @@ module('Integration | Identifiers - configuration', function (hooks) {
     this.owner.register('serializer:application', TestSerializer);
 
     let generateLidCalls = 0;
-    setIdentifierGenerationMethod((resource: ExistingResourceObject) => {
+    setIdentifierGenerationMethod((resource: ResourceData | { type: string }) => {
```

Comment:
It doesn't seem right that you accept `{ type: string }` but then error out if you get it

Reply:
This is because the implementation in this test expects to not be asked to generate a new identifier.